src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCard.component.ts

```diff
@@ -110,20 +124,30 @@ export class NotebookViewsCardComponent implements OnInit {
 		}
 	}
 
+	public set executionState(state: CellExecutionState) {
+		if (this._executionState !== state) {
+			this._executionState = state;
+		}
+	}
+
+	public get executionState(): CellExecutionState {
+		return this._executionState;
+	}
+
 	public hide(): void {
 		this.changed('hide');
 	}
 
 	public get data(): any {
```

Comment:
Can you give this an actual type? And wouldn't `metadata` be a better name here given that's what the actual underlying property is called?